web/src/main/java/org/mskcc/cbio/oncokb/api/pub/v1/TrialsApiController.java

```diff
@@ -1,242 +1,215 @@
+                        .getInstance()
+                        .getTrialsByIDList(tumor.getTrials()),
+                    treatment
+                );
+            return new ResponseEntity<List<ClinicalTrial>>(trial, status);
         }
-        return new ResponseEntity<List<Trial>>(new ArrayList<>(), status);
+        return new ResponseEntity<List<ClinicalTrial>>(
+            new ArrayList<>(),
+            status
+        );
     }
 
+    @PublicApi
     @PremiumPublicApi
-    @ApiOperation("Return a list of trials using cancer types")
-    @ApiResponses(value = {@ApiResponse(code = 200, message = "OK", responseContainer = "Map"),
-        @ApiResponse(code = 400, message = "Error", response = String.class)})
-    @RequestMapping(value = "/trials/cancerTypes", produces = {"application/json"}, method = RequestMethod.POST)
-    public ResponseEntity<Map<String, List<Trial>>> trialsGetByCancerTypes(
-        @ApiParam(value = "", required = true) @RequestBody() CancerTypesQuery body)
+    @ApiOperation(
+        "Return a list of clinical trials by cancer type, treatment, location and distance"
+    )
+    @ApiResponses(
+        value = {
+            @ApiResponse(code = 200, message = "OK", responseContainer = "Map"),
+            @ApiResponse(
+                code = 400,
+                message = "Error",
+                response = String.class
+            ),
+        }
+    )
+    @RequestMapping(
+        value = "/trials/cancerType",
+        produces = { "application/json" },
+        method = RequestMethod.GET
+    )
+    public ResponseEntity<List<ClinicalTrial>> trialsGetByCancerType(
+        @ApiParam(
+            value = "The cancer type that clinical trials belong to. Support cancer type name and OncoTree Code. Example: Glioma or AASTR. Support special cancer types: ALL_TUMORS, ALL_SOLID_TUMORS, ALL_LIQUID_TUMORS."
+        ) @RequestParam(
+            value = "cancerType",
+            required = true
+        ) String cancerType,
+        @ApiParam(
+            value = "Consisted of single/multuple drugs. Support drugs name or drugs' NCIT code. For multiple drugs treatment, use '+' as separator. Example: Everolimus+C95701"
+        ) @RequestParam(value = "treatment", required = false) String treatment,
+        @ApiParam(
+            value = "The address of your location. Support zip code. Must be spcified with country. Example: New York City, NY"
+        ) @RequestParam(value = "address", required = false) String address,
+        @ApiParam(
+            value = "The country of your location. Must be specified with address. Example: United States"
+        ) @RequestParam(value = "country", required = false) String country,
+        @ApiParam(
+            value = "The radius from your location. Must be specified with address and country. Example: 100, which means all trails have any site locates within 100 km from your location. If not specify, the default distance is 100km."
```

Comment:
These descriptions should be put into constant and share with annotation endpoints.